lib/utils.js

```diff
@@ -7,98 +10,79 @@ var assert = require("assert");
 
-  //user:password , host:port,host2:port2/databasename?option=someOption
-  var auth = auth_hosts_path_options.split("@")[0];
+  // The name comes from the path in the URL. We must always use the last host
+  function getName () {
+    const pathname = url.parse(parts[parts.length - 1]).pathname;
 
-
-  var hosts_path_options = auth_hosts_path_options.split("@")[1];
-
-  // host:port,host2:port2/databasename , option=someOption
-  var hosts_path = hosts_path_options.split("?")[0];
-  var options = hosts_path_options.split("?")[1];
-
-  //host:port,host2:port2, databasename
-  var hosts = hosts_path.split("/")[0];
-  var databaseName = hosts_path.split("/")[1];
-
-  //host:port host2:port2
-  var hostList = hosts.split(",");
-
-  //user, password
-  var authUser = auth.split(":")[0];
-  var authPassword = auth.split(":")[1];
-
-  result.database = {};
-  result.database.driver_options = {};
-  result.database.auth = {};
-  result.database.auth.user = authUser;
-  result.database.auth.pass = authPassword;
-  result.database.auth.source = databaseName;
-  result.database.name = databaseName;
-
-  assert.ok(authUser !== undefined);
-  assert.ok(authPassword !== undefined);
-  assert.ok(databaseName !== undefined);
-
-  //Parsing options
-  if (options) {
-    var parsedOptions = options.split(",");
-    //If options are parsed, they should have an even number of elements in the split array
-    assert.ok(parsedOptions.length > 0);
-
-    for (var i = 0; i < parsedOptions.length; i++) {
-      var splitOption = parsedOptions[i].split("=");
-      //Each option should be something=value
-      assert.ok(splitOption && splitOption.length === 2);
-      assert.ok(splitOption[0].length > 0);
-      assert.ok(splitOption[1].length > 0);
-      //Checking it does not already exist
-      assert.ok(!result.database.driver_options[splitOption[0]]);
-
-      result.database.driver_options[splitOption[0]] = splitOption[1];
-    }
+    return pathname ? pathname.replace('/', '') : pathname;
   }
 
-  var host_port;
-  if (hostList.length > 1) {
-    result.database.host = [];
-    result.database.port = [];
-    for (i = 0; i < hostList.length; i++) {
-      host_port = hostList[i].split(":");
-      assert.ok(host_port[0] && host_port[0].length > 0);
-      result.database.host.push(host_port[0]);
-      if (host_port.length > 1) {
-        result.database.port.push(host_port[1]);
-      } else {
-        result.database.port.push('27017');
-      }
+  // Parse the query string from the last host portion
+  function getOptionsObject () {
+    if (parts.length > 1) {
+      // Options should be appended to the last hosts querystring
+      return qs.parse(url.parse(parts[parts.length - 1]).query);
     }
+
+    return {};
   }
-  else {
-    host_port = hostList[0].split(":");
-    assert.ok(host_port[0] && host_port[0].length > 0);
-    result.database.host = host_port[0];
-    if (host_port.length > 1) {
-      result.database.port = host_port[1];
+  
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the useless trailing whitespaces at the end of this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)